### ISSUE TITLE
[Bug]: Ballerina shell crashes with java.lang.StringIndexOutOfBoundsException: String index out of range: 5 on rxvt-unicode terminal emulator

### ISSUE THREAD

**Issue by saper** [ballerina-platform/ballerina-lang#41019]
### Description

`bal shell` crashes at start when started on rxvt-unicode terminal emulator

```
ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in
https://github.com/ballerina-platform/ballerina-lang/issues with the
log you get below and your sample code.

We thank you for helping make us better.

[2023-07-15 03:21:43,405] SEVERE {b7a.log.crash} - java.lang.StringIndexOutOfBoundsException: String index out of range: 5 
java.io.IOError: java.lang.StringIndexOutOfBoundsException: String index out of range: 5
	at org.jline.utils.Curses.tputs(Curses.java:62)
	at org.jline.utils.Curses.tputs(Curses.java:45)
	at org.jline.keymap.KeyMap.key(KeyMap.java:243)
	at org.jline.reader.impl.LineReaderImpl.key(LineReaderImpl.java:5784)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.jline.reader.impl.LineReaderImpl.bindKeys(LineReaderImpl.java:5792)
	at org.jline.reader.impl.LineReaderImpl.emacs(LineReaderImpl.java:5503)
	at org.jline.reader.impl.LineReaderImpl.defaultKeyMaps(LineReaderImpl.java:5479)
	at org.jline.reader.impl.LineReaderImpl.<init>(LineReaderImpl.java:277)
	at org.jline.reader.LineReaderBuilder.build(LineReaderBuilder.java:115)
	at io.ballerina.shell.cli.ReplShellApplication.execute(ReplShellApplication.java:79)
	at io.ballerina.cli.cmd.ShellCommand.execute(ShellCommand.java:79)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.ballerina.cli.launcher.Main.main(Main.java:56)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 5
	at java.base/java.lang.StringLatin1.charAt(StringLatin1.java:47)
	at java.base/java.lang.String.charAt(String.java:693)
	at org.jline.utils.Curses.doTputs(Curses.java:351)
	at org.jline.utils.Curses.tputs(Curses.java:60)
	... 21 more

```
This seems to be https://github.com/jline/jline3/issues/533, fixed in [JLine 3.15.0](https://github.com/jline/jline3/issues/532)

### Steps to Reproduce

1. Open urxvt terminal, make sure `TERM=rxvt-unicode-256color`
2. ssh to the target
3. Verify that `TERM=rxvt-unicode-256color` and the terminal is known running `tput -T rxvt-unicode-256color longname` should return `rxvt-unicode terminal with 256 colors (X Window System)`
4. run `bal shell`

Expected:

```
Welcome to Ballerina Shell REPL.
Type /exit to exit and /help to list available commands.
```

### Affected Version(s)

ballerina-2201.7.0-swan-lake-linux-x64.rpm

### OS, DB, other environment details and versions

Running ballerina via ssh: AlmaLinux release 9.2 (Turquoise Kodkod)
Running the terminal: FreeBSD 14.0-CURRENT
Terminal emulator: rxvt-unicode 9.31 

### Related area

-> Shell

### Related issue(s) (optional)

https://github.com/jline/jline3/issues/533

**Comment by hasithaa** [comment]
Thanks @saper reporting this. I have created a PR updating the jline version. (https://github.com/ballerina-platform/ballerina-lang/pull/41222). Verified the fix with and without `TERM=rxvt-unicode-256color`

**Comment by lochana-chathura** [comment]
Fixed in #41222
